shapeshift/web
# feat: upgrade swapper protocol fee management

## Description

Unreverts #4528 with a fix to thorchain swaps with donations reported here 
https://discord.com/channels/554694662431178782/1110376655106605057/1110394152602063029

## Pull Request Type


## Issue (if applicable)

NA

## Risk

The majority of this chnage was reviewed, tested and approved in #4528, but for the fix there is moderate risk of incorrect fee values for thorchain swaps with/without donations enabled.

## Testing

- Check the issue reported is resolved
- Check protocol fees are still correct

### Engineering

See above

### Operations

Protocol fees, and exepected amount should be matching production

## Screenshots (if applicable)

with donation enabled
![image](https://github.com/shapeshift/web/assets/125113430/0406d590-c4a8-431e-a55b-bdcd4a4145fa)

with donation disabled
![image](https://github.com/shapeshift/web/assets/125113430/d3a1f989-2281-4cce-8fb7-6b3d16400447)
